GraphIntegrationTest.test_error_message is flaky in CI

See here: https://travis-ci.org/pantsbuild/pants/jobs/456235832

Looks like:
```
02:33:47 40:20       [run]
                     ============== test session starts ===============
                     platform linux -- Python 3.4.3, pytest-3.6.4, py-1.7.0, pluggy-0.7.1 -- /usr/bin/python3.4
                     cachedir: .pants.d/.pytest_cache
                     shard: 0 of 6 (0-based shard numbering)
                     rootdir: /home/travis/build/pantsbuild/pants/.pants.d, inifile: /home/travis/build/pantsbuild/pants/.pants.d/test/pytest-prep/CPython-3.4.3/8f4e2750fa5c8024cd5e6f1ef32266675b656ff3/pytest.ini
                     plugins: cov-2.4.0, timeout-1.2.1
                     collecting ... Only executing 1 of 5 total tests in shard 0 of 6
                     collected 5 items
                     
                     tests/python/pants_test/engine/legacy/test_graph_integration.py::GraphIntegrationTest::test_error_message <- pyprep/sources/d6c781ce5af299b6f6553cbdf77515274ffdf271/pants_test/engine/legacy/test_graph_integration.py FAILED [100%]
                     
                     ==================== FAILURES ====================
                     ____ GraphIntegrationTest.test_error_message _____
                     
                     self = <pants_test.engine.legacy.test_graph_integration.GraphIntegrationTest testMethod=test_error_message>
                     
                         def test_error_message(self):
                           for k in self._ERR_TARGETS:
                     >       self._list_target_check_error(k)
                     
                     .pants.d/pyprep/sources/d6c781ce5af299b6f6553cbdf77515274ffdf271/pants_test/engine/legacy/test_graph_integration.py:141: 
                     _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
                     .pants.d/pyprep/sources/d6c781ce5af299b6f6553cbdf77515274ffdf271/pants_test/engine/legacy/test_graph_integration.py:94: in _list_target_check_error
                         self.assertIn(excerpt, pants_run.stderr_data)
                     E   AssertionError: "Snapshot(PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/*.aaaa',), exclude=(), glob_match_error_behavior<=GlobMatchErrorBehavior>=GlobMatchErrorBehavior(failure_behavior=error), conjunction<=GlobExpansionConjunction>=GlobExpansionConjunction(conjunction=all_match)))" not found in 'timestamp: 2018-11-17T02:33:51.224568\nException caught: (pants.build_graph.address_lookup_error.AddressLookupError)\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 75, in <module>\n    main()\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 71, in main\n    PantsLoader.run()\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 67, in run\n    cls.load_and_execute(entrypoint)\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 60, in load_and_execute\n    entrypoint_main()\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_exe.py", line 39, in main\n    PantsRunner(exiter, start_time=start_time).run()\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_runner.py", line 62, in run\n    return runner.run()\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 159, in run\n    self._run()\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 227, in _run\n    goal_runner_result = self._maybe_run_v1(run_tracker, reporting)\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 176, in _maybe_run_v1\n    return goal_runner_factory.create().run()\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/goal_runner.py", line 125, in create\n    goals, context = self._setup_context()\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/goal_runner.py", line 95, in _setup_context\n    self._root_dir\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/init/engine_initializer.py", line 226, in create_build_graph\n    for _ in graph.inject_roots_closure(target_roots):\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 211, in inject_roots_closure\n    for address in self._inject_specs(target_roots.specs):\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 264, in _inject_specs\n    subjects)\n  File "/opt/pyenv/versions/2.7.14/lib/python2.7/contextlib.py", line 35, in __exit__\n    self.gen.throw(type, value, traceback)\n  File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 231, in _resolve_context\n    \'Build graph construction failed: {} {}\'.format(type(e).__name__, str(e))\n\nException message: Build graph construction failed: ExecutionError 1 Exception encountered:\nComputing Select(Specs(dependencies<=tuple>=(SingleAddress(directory=u\'testprojects/src/java/org/pantsbuild/testproject/bundle\', name=u\'missing-bundle-fileset\'),), matcher<=SpecsMatcher>=SpecsMatcher(tags<=tuple>=(), exclude_patterns<=tuple>=())), =TransitiveHydratedTargets)\n  Computing Task(transitive_hydrated_targets, Specs(dependencies<=tuple>=(SingleAddress(directory=u\'testprojects/src/java/org/pantsbuild/testproject/bundle\', name=u\'missing-bundle-fileset\'),), matcher<=SpecsMatcher>=SpecsMatcher(tags<=tuple>=(), exclude_patterns<=tuple>=())), =TransitiveHydratedTargets, true)\n    Computing Task(transitive_hydrated_target, testprojects/src/java/org/pantsbuild/testproject/bundle:missing-bundle-fileset, =TransitiveHydratedTarget, true)\n      Computing Task(hydrate_target, testprojects/src/java/org/pantsbuild/testproject/bundle:missing-bundle-fileset, =HydratedTarget, true)\n        Computing Task(hydrate_bundles, BundlesField(address=BuildFileAddress(testprojects/src/java/org/pantsbuild/testproject/bundle/BUILD, missing-bundle-fileset), bundles=[BundleAdaptor(fileset=[\'a/b/file1.txt\']), BundleAdaptor(fileset=RGlobs(\'*.aaaa\', \'*.bbbb\')), BundleAdaptor(fileset=Globs(\'*.aaaa\')), BundleAdaptor(fileset=ZGlobs(\'**/*.abab\')), BundleAdaptor(fileset=[\'file1.aaaa\', \'file2.aaaa\'])], filespecs_list=[{u\'globs\': [u\'a/b/file1.txt\']}, {u\'globs\': [u\'**/*.aaaa\', u\'**/*.bbbb\']}, {u\'globs\': [u\'*.aaaa\']}, {u\'globs\': [u\'**/*.abab\']}, {u\'globs\': [u\'file1.aaaa\', u\'file2.aaaa\']}], path_globs_list=[PathGlobs(include=(u\'testprojects/src/java/org/pantsbuild/testproject/bundle/a/b/file1.txt\',), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u\'ignore\'), conjunction=GlobExpansionConjunction(conjunction=u\'all_match\')), PathGlobs(include=(u\'testprojects/src/java/org/pantsbuild/testproject/bundle/**/*.aaaa\', u\'testprojects/src/java/org/pantsbuild/testproject/bundle/**/*.bbbb\'), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u\'ignore\'), conjunction=GlobExpansionConjunction(conjunction=u\'all_match\')), PathGlobs(include=(u\'testprojects/src/java/org/pantsbuild/testproject/bundle/*.aaaa\',), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u\'ignore\'), conjunction=GlobExpansionConjunction(conjunction=u\'all_match\')), PathGlobs(include=(u\'testprojects/src/java/org/pantsbuild/testproject/bundle/**/*.abab\',), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u\'ignore\'), conjunction=GlobExpansionConjunction(conjunction=u\'all_match\')), PathGlobs(include=(u\'testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa\', u\'testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa\'), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u\'ignore\'), conjunction=GlobExpansionConjunction(conjunction=u\'all_match\'))]), =HydratedField, true)\n          Computing Snapshot(PathGlobs(include=(u\'testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa\', u\'testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa\'), exclude=(), glob_match_error_behavior<=GlobMatchErrorBehavior>=GlobMatchErrorBehavior(failure_behavior=error), conjunction<=GlobExpansionConjunction>=GlobExpansionConjunction(conjunction=all_match)))\n            Throw(PathGlobs expansion failed: Throw(Globs did not match. Excludes were: []. Unmatched globs were: ["testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa", "testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa"]., "<pants native internals>"))\n              Traceback (no traceback):\n                <pants native internals>\n              Exception: PathGlobs expansion failed: Throw(Globs did not match. Excludes were: []. Unmatched globs were: ["testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa", "testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa"]., "<pants native internals>")\n\n\n\n'
                     -------------- Captured stdout call --------------
                     logs/exceptions.log +++ 
                     logs/exceptions.log >>> timestamp: 2018-11-17T02:33:49.811242
                     logs/exceptions.log >>> process title: python /home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py --no-pantsrc --pants-workdir=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmpzt6wb2wy.pants.d --kill-nailguns --print-exception-stacktrace=True --pants-config-files=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmpzt6wb2wy.pants.d/pants.ini list testprojects/src/python/sources:some-missing-some-not
                     logs/exceptions.log >>> sys.argv: ['/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py', '--no-pantsrc', '--pants-workdir=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmpzt6wb2wy.pants.d', '--kill-nailguns', '--print-exception-stacktrace=True', '--pants-config-files=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmpzt6wb2wy.pants.d/pants.ini', 'list', 'testprojects/src/python/sources:some-missing-some-not']
                     logs/exceptions.log >>> pid: 8666
                     logs/exceptions.log >>> Exception caught: (pants.build_graph.address_lookup_error.AddressLookupError)
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 75, in <module>
                     logs/exceptions.log >>> main()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 71, in main
                     logs/exceptions.log >>> PantsLoader.run()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 67, in run
                     logs/exceptions.log >>> cls.load_and_execute(entrypoint)
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 60, in load_and_execute
                     logs/exceptions.log >>> entrypoint_main()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_exe.py", line 39, in main
                     logs/exceptions.log >>> PantsRunner(exiter, start_time=start_time).run()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_runner.py", line 62, in run
                     logs/exceptions.log >>> return runner.run()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 159, in run
                     logs/exceptions.log >>> self._run()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 227, in _run
                     logs/exceptions.log >>> goal_runner_result = self._maybe_run_v1(run_tracker, reporting)
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 176, in _maybe_run_v1
                     logs/exceptions.log >>> return goal_runner_factory.create().run()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/goal_runner.py", line 125, in create
                     logs/exceptions.log >>> goals, context = self._setup_context()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/goal_runner.py", line 95, in _setup_context
                     logs/exceptions.log >>> self._root_dir
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/init/engine_initializer.py", line 226, in create_build_graph
                     logs/exceptions.log >>> for _ in graph.inject_roots_closure(target_roots):
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 211, in inject_roots_closure
                     logs/exceptions.log >>> for address in self._inject_specs(target_roots.specs):
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 264, in _inject_specs
                     logs/exceptions.log >>> subjects)
                     logs/exceptions.log >>> File "/opt/pyenv/versions/2.7.14/lib/python2.7/contextlib.py", line 35, in __exit__
                     logs/exceptions.log >>> self.gen.throw(type, value, traceback)
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 231, in _resolve_context
                     logs/exceptions.log >>> 'Build graph construction failed: {} {}'.format(type(e).__name__, str(e))
                     logs/exceptions.log >>> 
                     logs/exceptions.log >>> Exception message: Build graph construction failed: ExecutionError 1 Exception encountered:
                     logs/exceptions.log >>> Computing Select(Specs(dependencies<=tuple>=(SingleAddress(directory=u'testprojects/src/python/sources', name=u'some-missing-some-not'),), matcher<=SpecsMatcher>=SpecsMatcher(tags<=tuple>=(), exclude_patterns<=tuple>=())), =TransitiveHydratedTargets)
                     logs/exceptions.log >>> Computing Task(transitive_hydrated_targets, Specs(dependencies<=tuple>=(SingleAddress(directory=u'testprojects/src/python/sources', name=u'some-missing-some-not'),), matcher<=SpecsMatcher>=SpecsMatcher(tags<=tuple>=(), exclude_patterns<=tuple>=())), =TransitiveHydratedTargets, true)
                     logs/exceptions.log >>> Computing Task(transitive_hydrated_target, testprojects/src/python/sources:some-missing-some-not, =TransitiveHydratedTarget, true)
                     logs/exceptions.log >>> Computing Task(hydrate_target, testprojects/src/python/sources:some-missing-some-not, =HydratedTarget, true)
                     logs/exceptions.log >>> Computing Task(hydrate_sources, SourcesField(address=BuildFileAddress(testprojects/src/python/sources/BUILD, some-missing-some-not), input_globs=globs('*.txt', '*.rs'), arg=sources, filespecs={u'globs': [u'*.txt', u'*.rs']}), =HydratedField, true)
                     logs/exceptions.log >>> Computing Snapshot(PathGlobs(include=(u'testprojects/src/python/sources/*.txt', u'testprojects/src/python/sources/*.rs'), exclude=(), glob_match_error_behavior<=GlobMatchErrorBehavior>=GlobMatchErrorBehavior(failure_behavior=error), conjunction<=GlobExpansionConjunction>=GlobExpansionConjunction(conjunction=all_match)))
                     logs/exceptions.log >>> Throw(PathGlobs expansion failed: Throw(Globs did not match. Excludes were: []. Unmatched globs were: ["testprojects/src/python/sources/*.rs"]., "<pants native internals>"))
                     logs/exceptions.log >>> Traceback (no traceback):
                     logs/exceptions.log >>> <pants native internals>
                     logs/exceptions.log >>> Exception: PathGlobs expansion failed: Throw(Globs did not match. Excludes were: []. Unmatched globs were: ["testprojects/src/python/sources/*.rs"]., "<pants native internals>")
                     logs/exceptions.log >>> 
                     logs/exceptions.log >>> 
                     logs/exceptions.log --- 
                     logs/exceptions.8666.log +++ 
                     logs/exceptions.8666.log >>> timestamp: 2018-11-17T02:33:49.811242
                     logs/exceptions.8666.log >>> process title: python /home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py --no-pantsrc --pants-workdir=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmpzt6wb2wy.pants.d --kill-nailguns --print-exception-stacktrace=True --pants-config-files=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmpzt6wb2wy.pants.d/pants.ini list testprojects/src/python/sources:some-missing-some-not
                     logs/exceptions.8666.log >>> sys.argv: ['/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py', '--no-pantsrc', '--pants-workdir=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmpzt6wb2wy.pants.d', '--kill-nailguns', '--print-exception-stacktrace=True', '--pants-config-files=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmpzt6wb2wy.pants.d/pants.ini', 'list', 'testprojects/src/python/sources:some-missing-some-not']
                     logs/exceptions.8666.log >>> pid: 8666
                     logs/exceptions.8666.log >>> Exception caught: (pants.build_graph.address_lookup_error.AddressLookupError)
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 75, in <module>
                     logs/exceptions.8666.log >>> main()
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 71, in main
                     logs/exceptions.8666.log >>> PantsLoader.run()
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 67, in run
                     logs/exceptions.8666.log >>> cls.load_and_execute(entrypoint)
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 60, in load_and_execute
                     logs/exceptions.8666.log >>> entrypoint_main()
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_exe.py", line 39, in main
                     logs/exceptions.8666.log >>> PantsRunner(exiter, start_time=start_time).run()
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_runner.py", line 62, in run
                     logs/exceptions.8666.log >>> return runner.run()
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 159, in run
                     logs/exceptions.8666.log >>> self._run()
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 227, in _run
                     logs/exceptions.8666.log >>> goal_runner_result = self._maybe_run_v1(run_tracker, reporting)
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 176, in _maybe_run_v1
                     logs/exceptions.8666.log >>> return goal_runner_factory.create().run()
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/goal_runner.py", line 125, in create
                     logs/exceptions.8666.log >>> goals, context = self._setup_context()
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/goal_runner.py", line 95, in _setup_context
                     logs/exceptions.8666.log >>> self._root_dir
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/init/engine_initializer.py", line 226, in create_build_graph
                     logs/exceptions.8666.log >>> for _ in graph.inject_roots_closure(target_roots):
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 211, in inject_roots_closure
                     logs/exceptions.8666.log >>> for address in self._inject_specs(target_roots.specs):
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 264, in _inject_specs
                     logs/exceptions.8666.log >>> subjects)
                     logs/exceptions.8666.log >>> File "/opt/pyenv/versions/2.7.14/lib/python2.7/contextlib.py", line 35, in __exit__
                     logs/exceptions.8666.log >>> self.gen.throw(type, value, traceback)
                     logs/exceptions.8666.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 231, in _resolve_context
                     logs/exceptions.8666.log >>> 'Build graph construction failed: {} {}'.format(type(e).__name__, str(e))
                     logs/exceptions.8666.log >>> 
                     logs/exceptions.8666.log >>> Exception message: Build graph construction failed: ExecutionError 1 Exception encountered:
                     logs/exceptions.8666.log >>> Computing Select(Specs(dependencies<=tuple>=(SingleAddress(directory=u'testprojects/src/python/sources', name=u'some-missing-some-not'),), matcher<=SpecsMatcher>=SpecsMatcher(tags<=tuple>=(), exclude_patterns<=tuple>=())), =TransitiveHydratedTargets)
                     logs/exceptions.8666.log >>> Computing Task(transitive_hydrated_targets, Specs(dependencies<=tuple>=(SingleAddress(directory=u'testprojects/src/python/sources', name=u'some-missing-some-not'),), matcher<=SpecsMatcher>=SpecsMatcher(tags<=tuple>=(), exclude_patterns<=tuple>=())), =TransitiveHydratedTargets, true)
                     logs/exceptions.8666.log >>> Computing Task(transitive_hydrated_target, testprojects/src/python/sources:some-missing-some-not, =TransitiveHydratedTarget, true)
                     logs/exceptions.8666.log >>> Computing Task(hydrate_target, testprojects/src/python/sources:some-missing-some-not, =HydratedTarget, true)
                     logs/exceptions.8666.log >>> Computing Task(hydrate_sources, SourcesField(address=BuildFileAddress(testprojects/src/python/sources/BUILD, some-missing-some-not), input_globs=globs('*.txt', '*.rs'), arg=sources, filespecs={u'globs': [u'*.txt', u'*.rs']}), =HydratedField, true)
                     logs/exceptions.8666.log >>> Computing Snapshot(PathGlobs(include=(u'testprojects/src/python/sources/*.txt', u'testprojects/src/python/sources/*.rs'), exclude=(), glob_match_error_behavior<=GlobMatchErrorBehavior>=GlobMatchErrorBehavior(failure_behavior=error), conjunction<=GlobExpansionConjunction>=GlobExpansionConjunction(conjunction=all_match)))
                     logs/exceptions.8666.log >>> Throw(PathGlobs expansion failed: Throw(Globs did not match. Excludes were: []. Unmatched globs were: ["testprojects/src/python/sources/*.rs"]., "<pants native internals>"))
                     logs/exceptions.8666.log >>> Traceback (no traceback):
                     logs/exceptions.8666.log >>> <pants native internals>
                     logs/exceptions.8666.log >>> Exception: PathGlobs expansion failed: Throw(Globs did not match. Excludes were: []. Unmatched globs were: ["testprojects/src/python/sources/*.rs"]., "<pants native internals>")
                     logs/exceptions.8666.log >>> 
                     logs/exceptions.8666.log >>> 
                     logs/exceptions.8666.log --- 
                     logs/exceptions.8736.log +++ 
                     logs/exceptions.8736.log >>> timestamp: 2018-11-17T02:33:51.224216
                     logs/exceptions.8736.log >>> process title: python /home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py --no-pantsrc --pants-workdir=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmp1tqk_6o7.pants.d --kill-nailguns --print-exception-stacktrace=True --pants-config-files=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmp1tqk_6o7.pants.d/pants.ini list testprojects/src/java/org/pantsbuild/testproject/bundle:missing-bundle-fileset
                     logs/exceptions.8736.log >>> sys.argv: ['/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py', '--no-pantsrc', '--pants-workdir=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmp1tqk_6o7.pants.d', '--kill-nailguns', '--print-exception-stacktrace=True', '--pants-config-files=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmp1tqk_6o7.pants.d/pants.ini', 'list', 'testprojects/src/java/org/pantsbuild/testproject/bundle:missing-bundle-fileset']
                     logs/exceptions.8736.log >>> pid: 8736
                     logs/exceptions.8736.log >>> Exception caught: (pants.build_graph.address_lookup_error.AddressLookupError)
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 75, in <module>
                     logs/exceptions.8736.log >>> main()
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 71, in main
                     logs/exceptions.8736.log >>> PantsLoader.run()
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 67, in run
                     logs/exceptions.8736.log >>> cls.load_and_execute(entrypoint)
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 60, in load_and_execute
                     logs/exceptions.8736.log >>> entrypoint_main()
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_exe.py", line 39, in main
                     logs/exceptions.8736.log >>> PantsRunner(exiter, start_time=start_time).run()
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_runner.py", line 62, in run
                     logs/exceptions.8736.log >>> return runner.run()
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 159, in run
                     logs/exceptions.8736.log >>> self._run()
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 227, in _run
                     logs/exceptions.8736.log >>> goal_runner_result = self._maybe_run_v1(run_tracker, reporting)
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 176, in _maybe_run_v1
                     logs/exceptions.8736.log >>> return goal_runner_factory.create().run()
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/goal_runner.py", line 125, in create
                     logs/exceptions.8736.log >>> goals, context = self._setup_context()
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/goal_runner.py", line 95, in _setup_context
                     logs/exceptions.8736.log >>> self._root_dir
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/init/engine_initializer.py", line 226, in create_build_graph
                     logs/exceptions.8736.log >>> for _ in graph.inject_roots_closure(target_roots):
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 211, in inject_roots_closure
                     logs/exceptions.8736.log >>> for address in self._inject_specs(target_roots.specs):
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 264, in _inject_specs
                     logs/exceptions.8736.log >>> subjects)
                     logs/exceptions.8736.log >>> File "/opt/pyenv/versions/2.7.14/lib/python2.7/contextlib.py", line 35, in __exit__
                     logs/exceptions.8736.log >>> self.gen.throw(type, value, traceback)
                     logs/exceptions.8736.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 231, in _resolve_context
                     logs/exceptions.8736.log >>> 'Build graph construction failed: {} {}'.format(type(e).__name__, str(e))
                     logs/exceptions.8736.log >>> 
                     logs/exceptions.8736.log >>> Exception message: Build graph construction failed: ExecutionError 1 Exception encountered:
                     logs/exceptions.8736.log >>> Computing Select(Specs(dependencies<=tuple>=(SingleAddress(directory=u'testprojects/src/java/org/pantsbuild/testproject/bundle', name=u'missing-bundle-fileset'),), matcher<=SpecsMatcher>=SpecsMatcher(tags<=tuple>=(), exclude_patterns<=tuple>=())), =TransitiveHydratedTargets)
                     logs/exceptions.8736.log >>> Computing Task(transitive_hydrated_targets, Specs(dependencies<=tuple>=(SingleAddress(directory=u'testprojects/src/java/org/pantsbuild/testproject/bundle', name=u'missing-bundle-fileset'),), matcher<=SpecsMatcher>=SpecsMatcher(tags<=tuple>=(), exclude_patterns<=tuple>=())), =TransitiveHydratedTargets, true)
                     logs/exceptions.8736.log >>> Computing Task(transitive_hydrated_target, testprojects/src/java/org/pantsbuild/testproject/bundle:missing-bundle-fileset, =TransitiveHydratedTarget, true)
                     logs/exceptions.8736.log >>> Computing Task(hydrate_target, testprojects/src/java/org/pantsbuild/testproject/bundle:missing-bundle-fileset, =HydratedTarget, true)
                     logs/exceptions.8736.log >>> Computing Task(hydrate_bundles, BundlesField(address=BuildFileAddress(testprojects/src/java/org/pantsbuild/testproject/bundle/BUILD, missing-bundle-fileset), bundles=[BundleAdaptor(fileset=['a/b/file1.txt']), BundleAdaptor(fileset=RGlobs('*.aaaa', '*.bbbb')), BundleAdaptor(fileset=Globs('*.aaaa')), BundleAdaptor(fileset=ZGlobs('**/*.abab')), BundleAdaptor(fileset=['file1.aaaa', 'file2.aaaa'])], filespecs_list=[{u'globs': [u'a/b/file1.txt']}, {u'globs': [u'**/*.aaaa', u'**/*.bbbb']}, {u'globs': [u'*.aaaa']}, {u'globs': [u'**/*.abab']}, {u'globs': [u'file1.aaaa', u'file2.aaaa']}], path_globs_list=[PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/a/b/file1.txt',), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u'ignore'), conjunction=GlobExpansionConjunction(conjunction=u'all_match')), PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/**/*.aaaa', u'testprojects/src/java/org/pantsbuild/testproject/bundle/**/*.bbbb'), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u'ignore'), conjunction=GlobExpansionConjunction(conjunction=u'all_match')), PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/*.aaaa',), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u'ignore'), conjunction=GlobExpansionConjunction(conjunction=u'all_match')), PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/**/*.abab',), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u'ignore'), conjunction=GlobExpansionConjunction(conjunction=u'all_match')), PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa', u'testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa'), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u'ignore'), conjunction=GlobExpansionConjunction(conjunction=u'all_match'))]), =HydratedField, true)
                     logs/exceptions.8736.log >>> Computing Snapshot(PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa', u'testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa'), exclude=(), glob_match_error_behavior<=GlobMatchErrorBehavior>=GlobMatchErrorBehavior(failure_behavior=error), conjunction<=GlobExpansionConjunction>=GlobExpansionConjunction(conjunction=all_match)))
                     logs/exceptions.8736.log >>> Throw(PathGlobs expansion failed: Throw(Globs did not match. Excludes were: []. Unmatched globs were: ["testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa", "testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa"]., "<pants native internals>"))
                     logs/exceptions.8736.log >>> Traceback (no traceback):
                     logs/exceptions.8736.log >>> <pants native internals>
                     logs/exceptions.8736.log >>> Exception: PathGlobs expansion failed: Throw(Globs did not match. Excludes were: []. Unmatched globs were: ["testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa", "testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa"]., "<pants native internals>")
                     logs/exceptions.8736.log >>> 
                     logs/exceptions.8736.log >>> 
                     logs/exceptions.8736.log --- 
                     logs/exceptions.log +++ 
                     logs/exceptions.log >>> timestamp: 2018-11-17T02:33:51.224216
                     logs/exceptions.log >>> process title: python /home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py --no-pantsrc --pants-workdir=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmp1tqk_6o7.pants.d --kill-nailguns --print-exception-stacktrace=True --pants-config-files=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmp1tqk_6o7.pants.d/pants.ini list testprojects/src/java/org/pantsbuild/testproject/bundle:missing-bundle-fileset
                     logs/exceptions.log >>> sys.argv: ['/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py', '--no-pantsrc', '--pants-workdir=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmp1tqk_6o7.pants.d', '--kill-nailguns', '--print-exception-stacktrace=True', '--pants-config-files=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmp1tqk_6o7.pants.d/pants.ini', 'list', 'testprojects/src/java/org/pantsbuild/testproject/bundle:missing-bundle-fileset']
                     logs/exceptions.log >>> pid: 8736
                     logs/exceptions.log >>> Exception caught: (pants.build_graph.address_lookup_error.AddressLookupError)
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 75, in <module>
                     logs/exceptions.log >>> main()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 71, in main
                     logs/exceptions.log >>> PantsLoader.run()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 67, in run
                     logs/exceptions.log >>> cls.load_and_execute(entrypoint)
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_loader.py", line 60, in load_and_execute
                     logs/exceptions.log >>> entrypoint_main()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_exe.py", line 39, in main
                     logs/exceptions.log >>> PantsRunner(exiter, start_time=start_time).run()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/pants_runner.py", line 62, in run
                     logs/exceptions.log >>> return runner.run()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 159, in run
                     logs/exceptions.log >>> self._run()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 227, in _run
                     logs/exceptions.log >>> goal_runner_result = self._maybe_run_v1(run_tracker, reporting)
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/local_pants_runner.py", line 176, in _maybe_run_v1
                     logs/exceptions.log >>> return goal_runner_factory.create().run()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/goal_runner.py", line 125, in create
                     logs/exceptions.log >>> goals, context = self._setup_context()
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/bin/goal_runner.py", line 95, in _setup_context
                     logs/exceptions.log >>> self._root_dir
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/init/engine_initializer.py", line 226, in create_build_graph
                     logs/exceptions.log >>> for _ in graph.inject_roots_closure(target_roots):
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 211, in inject_roots_closure
                     logs/exceptions.log >>> for address in self._inject_specs(target_roots.specs):
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 264, in _inject_specs
                     logs/exceptions.log >>> subjects)
                     logs/exceptions.log >>> File "/opt/pyenv/versions/2.7.14/lib/python2.7/contextlib.py", line 35, in __exit__
                     logs/exceptions.log >>> self.gen.throw(type, value, traceback)
                     logs/exceptions.log >>> File "/home/travis/build/pantsbuild/pants/src/python/pants/engine/legacy/graph.py", line 231, in _resolve_context
                     logs/exceptions.log >>> 'Build graph construction failed: {} {}'.format(type(e).__name__, str(e))
                     logs/exceptions.log >>> 
                     logs/exceptions.log >>> Exception message: Build graph construction failed: ExecutionError 1 Exception encountered:
                     logs/exceptions.log >>> Computing Select(Specs(dependencies<=tuple>=(SingleAddress(directory=u'testprojects/src/java/org/pantsbuild/testproject/bundle', name=u'missing-bundle-fileset'),), matcher<=SpecsMatcher>=SpecsMatcher(tags<=tuple>=(), exclude_patterns<=tuple>=())), =TransitiveHydratedTargets)
                     logs/exceptions.log >>> Computing Task(transitive_hydrated_targets, Specs(dependencies<=tuple>=(SingleAddress(directory=u'testprojects/src/java/org/pantsbuild/testproject/bundle', name=u'missing-bundle-fileset'),), matcher<=SpecsMatcher>=SpecsMatcher(tags<=tuple>=(), exclude_patterns<=tuple>=())), =TransitiveHydratedTargets, true)
                     logs/exceptions.log >>> Computing Task(transitive_hydrated_target, testprojects/src/java/org/pantsbuild/testproject/bundle:missing-bundle-fileset, =TransitiveHydratedTarget, true)
                     logs/exceptions.log >>> Computing Task(hydrate_target, testprojects/src/java/org/pantsbuild/testproject/bundle:missing-bundle-fileset, =HydratedTarget, true)
                     logs/exceptions.log >>> Computing Task(hydrate_bundles, BundlesField(address=BuildFileAddress(testprojects/src/java/org/pantsbuild/testproject/bundle/BUILD, missing-bundle-fileset), bundles=[BundleAdaptor(fileset=['a/b/file1.txt']), BundleAdaptor(fileset=RGlobs('*.aaaa', '*.bbbb')), BundleAdaptor(fileset=Globs('*.aaaa')), BundleAdaptor(fileset=ZGlobs('**/*.abab')), BundleAdaptor(fileset=['file1.aaaa', 'file2.aaaa'])], filespecs_list=[{u'globs': [u'a/b/file1.txt']}, {u'globs': [u'**/*.aaaa', u'**/*.bbbb']}, {u'globs': [u'*.aaaa']}, {u'globs': [u'**/*.abab']}, {u'globs': [u'file1.aaaa', u'file2.aaaa']}], path_globs_list=[PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/a/b/file1.txt',), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u'ignore'), conjunction=GlobExpansionConjunction(conjunction=u'all_match')), PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/**/*.aaaa', u'testprojects/src/java/org/pantsbuild/testproject/bundle/**/*.bbbb'), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u'ignore'), conjunction=GlobExpansionConjunction(conjunction=u'all_match')), PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/*.aaaa',), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u'ignore'), conjunction=GlobExpansionConjunction(conjunction=u'all_match')), PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/**/*.abab',), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u'ignore'), conjunction=GlobExpansionConjunction(conjunction=u'all_match')), PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa', u'testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa'), exclude=(), glob_match_error_behavior=GlobMatchErrorBehavior(failure_behavior=u'ignore'), conjunction=GlobExpansionConjunction(conjunction=u'all_match'))]), =HydratedField, true)
                     logs/exceptions.log >>> Computing Snapshot(PathGlobs(include=(u'testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa', u'testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa'), exclude=(), glob_match_error_behavior<=GlobMatchErrorBehavior>=GlobMatchErrorBehavior(failure_behavior=error), conjunction<=GlobExpansionConjunction>=GlobExpansionConjunction(conjunction=all_match)))
                     logs/exceptions.log >>> Throw(PathGlobs expansion failed: Throw(Globs did not match. Excludes were: []. Unmatched globs were: ["testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa", "testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa"]., "<pants native internals>"))
                     logs/exceptions.log >>> Traceback (no traceback):
                     logs/exceptions.log >>> <pants native internals>
                     logs/exceptions.log >>> Exception: PathGlobs expansion failed: Throw(Globs did not match. Excludes were: []. Unmatched globs were: ["testprojects/src/java/org/pantsbuild/testproject/bundle/file1.aaaa", "testprojects/src/java/org/pantsbuild/testproject/bundle/file2.aaaa"]., "<pants native internals>")
                     logs/exceptions.log >>> 
                     logs/exceptions.log >>> 
                     logs/exceptions.log --- 
                      generated xml file: /home/travis/build/pantsbuild/pants/.pants.d/test/pytest/tests.python.pants_test.engine.legacy.graph_integration/junitxml/TEST-tests.python.pants_test.engine.legacy.graph_integration.xml 
                     ============ slowest 3 test durations ============
                     2.80s call     ../tests/python/pants_test/engine/legacy/test_graph_integration.py::GraphIntegrationTest::test_error_message
                     0.00s teardown ../tests/python/pants_test/engine/legacy/test_graph_integration.py::GraphIntegrationTest::test_error_message
                     0.00s setup    ../tests/python/pants_test/engine/legacy/test_graph_integration.py::GraphIntegrationTest::test_error_message
                     ============ 1 failed in 2.92 seconds ============
```

--- a/src/rust/engine/fs/Cargo.toml
+++ b/src/rust/engine/fs/Cargo.toml
@@ -17,7 +17,7 @@ glob = "0.2.11"
 grpcio = { git = "https://github.com/pantsbuild/grpc-rs.git", rev = "4dfafe9355dc996d7d0702e7386a6fedcd9734c0", default_features = false, features = ["protobuf-codec"] }
 hashing = { path = "../hashing" }
 ignore = "0.4.4"
-indexmap = "1"
+indexmap = "1.0.2"
 itertools = "0.7.2"
 lazy_static = "1"
 lmdb = { git = "https://github.com/pantsbuild/lmdb-rs.git", rev = "06bdfbfc6348f6804127176e561843f214fc17f8" }

--- a/src/rust/engine/graph/Cargo.toml
+++ b/src/rust/engine/graph/Cargo.toml
@@ -9,6 +9,7 @@ boxfuture = { path = "../boxfuture" }
 fnv = "1.0.5"
 futures = "^0.1.16"
 hashing = { path = "../hashing" }
+indexmap = "1.0.2"
 parking_lot = "0.6"
 petgraph = "0.4.5"
 

--- a/src/rust/engine/graph/src/lib.rs
+++ b/src/rust/engine/graph/src/lib.rs
@@ -34,6 +34,7 @@ extern crate boxfuture;
 extern crate fnv;
 extern crate futures;
 extern crate hashing;
+extern crate indexmap;
 extern crate parking_lot;
 extern crate petgraph;
 
@@ -54,6 +55,7 @@ use std::time::{Duration, Instant};
 use fnv::FnvHasher;
 
 use futures::future::{self, Future};
+use indexmap::IndexSet;
 use parking_lot::Mutex;
 use petgraph::graph::DiGraph;
 use petgraph::visit::EdgeRef;
@@ -270,7 +272,7 @@ impl<N: Node> InnerGraph<N> {
   }
 
   fn trace<T: NodeTracer<N>>(&self, roots: &[N], file_path: &Path) -> Result<(), String> {
-    let root_ids: HashSet<EntryId, FNV> = roots
+    let root_ids: IndexSet<EntryId, FNV> = roots
       .into_iter()
       .filter_map(|nk| self.entry_id(&EntryKey::Valid(nk.clone())))
       .cloned()

